File: openbis_api/source/java/ch/ethz/sis/openbis/generic/dssapi/v3/dto/dataset/create/UploadedDataSetCreation.java
package ch.ethz.sis.openbis.generic.dssapi.v3.dto.dataset.create;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.id.IDataSetId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.entitytype.id.IEntityTypeId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.experiment.id.IExperimentId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.sample.id.ISampleId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("dss.dto.dataset.create.UploadedDataSetCreation")
public class UploadedDataSetCreation implements ICreation
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private IEntityTypeId typeId;

    @JsonProperty
    private IExperimentId experimentId;

    @JsonProperty
    private ISampleId sampleId;

    @JsonProperty
    private Map<String, String> properties = new HashMap<String, String>();

    @JsonProperty
    private List<? extends IDataSetId> parentIds;

    @JsonProperty
    private String uploadId;

    public IEntityTypeId getTypeId()
    {
        return typeId;
    }

    public void setTypeId(IEntityTypeId typeId)
    {
        this.typeId = typeId;
    }

    public IExperimentId getExperimentId()
    {
        return experimentId;
    }

    public void setExperimentId(IExperimentId experimentId)
    {
        this.experimentId = experimentId;
    }

    public ISampleId getSampleId()
    {
        return sampleId;
    }

    public void setSampleId(ISampleId sampleId)
    {
        this.sampleId = sampleId;
    }

    public Map<String, String> getProperties()
    {
        return properties;
    }

    public void setProperties(Map<String, String> properties)
    {
        this.properties = properties;
    }

    public List<? extends IDataSetId> getParentIds()
    {
        return parentIds;
    }

    public void setParentIds(List<? extends IDataSetId> parentIds)
    {
        this.parentIds = parentIds;
    }

    public String getUploadId()
    {
        return uploadId;
    }

    public void setUploadId(String uploadId)
    {
        this.uploadId = uploadId;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dss/dto/dataset/create/UploadedDataSetCreation.js
define([ "stjs" ], function(stjs) {
	var UploadedDataSetCreation = function() {
		this.properties = {};
	};
	stjs.extend(UploadedDataSetCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'dss.dto.dataset.create.UploadedDataSetCreation';
		constructor.serialVersionUID = 1;
		prototype.typeId = null;
		prototype.experimentId = null;
		prototype.sampleId = null;
		prototype.properties = null;
		prototype.parentIds = null;
		prototype.uploadId = null;

		prototype.getTypeId = function() {
			return this.typeId;
		};
		prototype.setTypeId = function(typeId) {
			this.typeId = typeId;
		};
		prototype.getExperimentId = function() {
			return this.experimentId;
		};
		prototype.setExperimentId = function(experimentId) {
			this.experimentId = experimentId;
		};
		prototype.getSampleId = function() {
			return this.sampleId;
		};
		prototype.setSampleId = function(sampleId) {
			this.sampleId = sampleId;
		};
		prototype.getProperty = function(propertyName) {
			return this.properties[propertyName];
		};
		prototype.setProperty = function(propertyName, propertyValue) {
			this.properties[propertyName] = propertyValue;
		};
		prototype.getProperties = function() {
			return this.properties;
		};
		prototype.setProperties = function(properties) {
			this.properties = properties;
		};
		prototype.getParentIds = function() {
			return this.parentIds;
		};
		prototype.setParentIds = function(parentIds) {
			this.parentIds = parentIds;
		};
		prototype.getUploadId = function() {
			return this.uploadId;
		};
		prototype.setUploadId = function(uploadId) {
			this.uploadId = uploadId;
		};
	}, {
		typeId : "IEntityTypeId",
		experimentId : "IExperimentId",
		sampleId : "ISampleId",
		properties : {
			name : "Map",
			arguments : [ null, null ]
		},
		parentIds : {
			name : "List",
			arguments : [ "Object" ]
		}
	});
	return UploadedDataSetCreation;
})